### PR TITLE
Jgrgicevic/res detail page

### PR DESCRIPTION
**Pull request by gjuro87**

### CODE REVIEW

File: frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const generateChartData = (resources) => {
+    const chartdata = []
+    for (const resource of resources) {
+      chartdata.push({
+        name: resource.version,
+        value: resource.versionList.edges.length
+      })
+    }
+    return chartdata
+  }
```
Comment (blfpd) [inline frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx]:
```js
const generateChartData = (resources) => resources.map(resource => ({
    name: resource.version,
    value: resource.versionList.edges.length
})
```

File: frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    function randomHSL(){
+      return `hsla(${~~(360 * Math.random())},70%,70%,0.8)`
+    }
+
+    const colors = []
+    for (let i = 0; i < chartData.length; i++) {
+      colors.push(randomHSL())
+    }
+
+    return colors
```
Comment (blfpd) [inline frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx]:
```js
return Array.from({length: chartData.length}, () => randoHSL())
```
Ref: [MDN: Array.from(), using arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#using_arrow_functions_and_array.from)

File: frontend/plugins/services/src/components/App.tsx [hunk 3]
```diff
@@ -19,7 +19,8 @@ export const App = () => (
             <Route path="/" element={<ServiceList />} />
             <Route path=":id/:name" element={<ServiceDetail />}>
               <Route path="swagger-ui" element={<ServiceApiDoc />} />
-              <Route path="/" element={<ServiceOverview />} />
+              <Route path="/" element={<
+                ServiceOverview />} />
```
Comment (blfpd) [inline frontend/plugins/services/src/components/App.tsx]: _upss_

File: frontend/plugins/resources/src/components/DetailPage/VersionDistribution.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            <Cell key={`cell-${index}`} fill={chartColors[index % chartColors.length]} />
+            ))}
+        </Pie>
+      </PieChart>
+    </ResponsiveContainer>
+
+  );
+
+
+}
```
Comment (blfpd) [inline frontend/plugins/resources/src/components/DetailPage/VersionDistribution.tsx:84]: Too much new lines in this file.

**Review by blfpd — CHANGES_REQUESTED** [review]

File: frontend/plugins/resources/src/components/DetailPage/VersionDistribution.tsx (hunk 4 above)
Comment (gjuro87) [reply frontend/plugins/resources/src/components/DetailPage/VersionDistribution.tsx:84]: fixed

File: frontend/plugins/services/src/components/App.tsx (hunk 3 above)
Comment (gjuro87) [reply frontend/plugins/services/src/components/App.tsx]: Yea, upss. Fixed.

File: frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx (hunk 2 above)
Comment (gjuro87) [reply frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx]: This looks nicer, tnx.

File: frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx (hunk 1 above)
Comment (gjuro87) [reply frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx]: This is better way, tnx.

**Pull request merged**